### PR TITLE
refactor: resolve Circular dependency

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

vite、rollup、microbundle 踩过多次坑。

close https://github.com/ant-design/ant-design/issues/42666
close https://github.com/ant-design/ant-design/issues/27209
close https://github.com/ant-design/ant-design/issues/41444
close https://github.com/ant-design/ant-design/issues/32094
close https://github.com/ant-design/ant-design/issues/18999
close https://github.com/ant-design/ant-design/issues/17803
close https://github.com/ant-design/ant-design/issues/5264

### 💡 Background and solution

问题太多了，大家有空慢慢来改。。

**📢 注意：（PR 目标分支为 feature）由于涉及改动模块过多，为避免领取了重复任务做无用功，感兴趣的同学可以在评论区留言认领任务，AntDesign 的 member 可以直接修改 ISSUE 详情中认领的任务后面加上 用户id 和 PR 链接，完成后可勾选完成任务。相关格式参考如下：**
image

https://github.com/ant-design/ant-design/actions/runs/5130129159/jobs/9228513045?pr=42750

- [x] (undefined) Circular dependency detected:
components/alert/ErrorBoundary.tsx -> components/alert/index.tsx -> components/alert/ErrorBoundary.tsx @jrr997 #42801
- [x] (undefined) Circular dependency detected:
components/alert/index.tsx -> components/alert/ErrorBoundary.tsx -> components/alert/index.tsx  @jrr997 #42801
- [x] (undefined) Circular dependency detected:
components/checkbox/Checkbox.tsx -> components/checkbox/Group.tsx -> components/checkbox/Checkbox.tsx @jrr997 #42803
- [x] (undefined) Circular dependency detected:
components/checkbox/Group.tsx -> components/checkbox/Checkbox.tsx -> components/checkbox/Group.tsx @jrr997 #42803
- [x] (undefined) Circular dependency detected:
components/color-picker/color.ts -> components/color-picker/util.ts -> components/color-picker/color.ts @MuxinFeng #42843
- [x] (undefined) Circular dependency detected:
components/color-picker/util.ts -> components/color-picker/color.ts -> components/color-picker/util.ts @MuxinFeng #42843
- [x] (undefined) Circular dependency detected:
components/config-provider/SizeContext.tsx -> components/config-provider/hooks/useSize.ts -> components/config-provider/SizeContext.tsx @jrr997 #42810
- [x] (undefined) Circular dependency detected:
components/config-provider/hooks/useSize.ts -> components/config-provider/SizeContext.tsx -> components/config-provider/hooks/useSize.ts @jrr997 #42810
- [x] (undefined) Circular dependency detected:
components/date-picker/generatePicker/generateRangePicker.tsx -> components/date-picker/generatePicker/index.tsx -> components/date-picker/generatePicker/generateRangePicker.tsx @jrr997 #42809
- [x] (undefined) Circular dependency detected:
components/date-picker/generatePicker/generateSinglePicker.tsx -> components/date-picker/generatePicker/index.tsx -> components/date-picker/generatePicker/generateSinglePicker.tsx @jrr997 #42809
- [x] (undefined) Circular dependency detected:
components/date-picker/generatePicker/index.tsx -> components/date-picker/generatePicker/generateRangePicker.tsx -> components/date-picker/generatePicker/index.tsx @jrr997 #42809
- [x] (undefined) Circular dependency detected:
components/descriptions/Row.tsx -> components/descriptions/index.tsx -> components/descriptions/Row.tsx @jrr997 #42812
- [x] (undefined) Circular dependency detected:
components/descriptions/index.tsx -> components/descriptions/Row.tsx -> components/descriptions/index.tsx @jrr997 #42812
- [x] (undefined) Circular dependency detected:
components/list/Item.tsx -> components/list/index.tsx -> components/list/Item.tsx @kiner-tang #42806
- [x] (undefined) Circular dependency detected:
components/list/index.tsx -> components/list/Item.tsx -> components/list/index.tsx @kiner-tang #42806
- [x] (undefined) Circular dependency detected:
components/modal/ConfirmDialog.tsx -> components/modal/Modal.tsx -> components/modal/PurePanel.tsx -> components/modal/ConfirmDialog.tsx @kiner-tang #42808
- [x] (undefined) Circular dependency detected:
components/modal/Modal.tsx -> components/modal/PurePanel.tsx -> components/modal/ConfirmDialog.tsx -> components/modal/Modal.tsx @kiner-tang #42808
- [x] (undefined) Circular dependency detected:
components/modal/PurePanel.tsx -> components/modal/ConfirmDialog.tsx -> components/modal/Modal.tsx -> components/modal/PurePanel.tsx @kiner-tang #42808
- [x] (undefined) Circular dependency detected:
components/space/Item.tsx -> components/space/index.tsx -> components/space/Item.tsx @kiner-tang #42811
- [x] (undefined) Circular dependency detected:
components/space/index.tsx -> components/space/Item.tsx -> components/space/index.tsx @kiner-tang #42811
- [x] (undefined) Circular dependency detected:
components/statistic/Countdown.tsx -> components/statistic/Statistic.tsx -> components/statistic/Countdown.tsx @kiner-tang #42814
- [x] (undefined) Circular dependency detected:
components/statistic/Statistic.tsx -> components/statistic/Countdown.tsx -> components/statistic/Statistic.tsx @kiner-tang #42814
- [x] (undefined) Circular dependency detected:
components/table/hooks/useFilter/FilterDropdown.tsx -> components/table/hooks/useFilter/index.tsx -> components/table/hooks/useFilter/FilterDropdown.tsx @jrr997 #42813
- [x] (undefined) Circular dependency detected:
components/table/hooks/useFilter/index.tsx -> components/table/hooks/useFilter/FilterDropdown.tsx -> components/table/hooks/useFilter/index.tsx @jrr997 #42813
- [x] (undefined) Circular dependency detected:
components/theme/internal.ts -> components/theme/util/genComponentStyleHook.ts -> components/theme/internal.ts @kiner-tang #42852
- [x] (undefined) Circular dependency detected:
components/theme/util/genComponentStyleHook.ts -> components/theme/internal.ts -> components/theme/util/genComponentStyleHook.ts @kiner-tang #42852
- [X] (undefined) Circular dependency detected:
node_modules/@rc-component/color-picker/es/color.js -> node_modules/@rc-component/color-picker/es/util.js -> node_modules/@rc-component/color-picker/es/color.js @linxianxi https://github.com/react-component/color-picker/commit/1f05f332045cc9cef0255c66f9c4f31fac2a00c5
- [X] (undefined) Circular dependency detected:
node_modules/@rc-component/color-picker/es/util.js -> node_modules/@rc-component/color-picker/es/color.js -> node_modules/@rc-component/color-picker/es/util.js @linxianxi https://github.com/react-component/color-picker/commit/1f05f332045cc9cef0255c66f9c4f31fac2a00c5 
- [x] (undefined) Circular dependency detected:
node_modules/rc-field-form/es/index.js -> node_modules/rc-field-form/es/useWatch.js -> node_modules/rc-field-form/es/index.js @kiner-tang https://github.com/react-component/field-form/pull/595
- [x] (undefined) Circular dependency detected:
node_modules/rc-field-form/es/useWatch.js -> node_modules/rc-field-form/es/index.js -> node_modules/rc-field-form/es/useWatch.js @kiner-tang https://github.com/react-component/field-form/pull/595
- [x] (undefined) Circular dependency detected:
node_modules/rc-image/es/Image.js -> node_modules/rc-image/es/PreviewGroup.js -> node_modules/rc-image/es/hooks/usePreviewItems.js -> node_modules/rc-image/es/Image.js @kiner-tang https://github.com/react-component/image/pull/262
- [x] (undefined) Circular dependency detected:
node_modules/rc-image/es/PreviewGroup.js -> node_modules/rc-image/es/hooks/usePreviewItems.js -> node_modules/rc-image/es/Image.js -> node_modules/rc-image/es/PreviewGroup.js @kiner-tang https://github.com/react-component/image/pull/262
- [x] (undefined) Circular dependency detected:
node_modules/rc-image/es/hooks/usePreviewItems.js -> node_modules/rc-image/es/Image.js -> node_modules/rc-image/es/PreviewGroup.js -> node_modules/rc-image/es/hooks/usePreviewItems.js @kiner-tang https://github.com/react-component/image/pull/262
- [x] (undefined) Circular dependency detected:
node_modules/rc-menu/es/Menu.js -> node_modules/rc-menu/es/SubMenu/index.js -> node_modules/rc-menu/es/utils/nodeUtil.js -> node_modules/rc-menu/es/index.js -> node_modules/rc-menu/es/Menu.js @kiner-tang https://github.com/react-component/menu/pull/655
- [x] (undefined) Circular dependency detected:
node_modules/rc-menu/es/MenuItemGroup.js -> node_modules/rc-menu/es/utils/nodeUtil.js -> node_modules/rc-menu/es/index.js -> node_modules/rc-menu/es/MenuItemGroup.js @kiner-tang https://github.com/react-component/menu/pull/655
- [x] (undefined) Circular dependency detected:
node_modules/rc-menu/es/SubMenu/index.js -> node_modules/rc-menu/es/utils/nodeUtil.js -> node_modules/rc-menu/es/index.js -> node_modules/rc-menu/es/Menu.js -> node_modules/rc-menu/es/SubMenu/index.js @kiner-tang https://github.com/react-component/menu/pull/655
- [x] (undefined) Circular dependency detected:
node_modules/rc-menu/es/index.js -> node_modules/rc-menu/es/Menu.js -> node_modules/rc-menu/es/SubMenu/index.js -> node_modules/rc-menu/es/utils/nodeUtil.js -> node_modules/rc-menu/es/index.js @kiner-tang https://github.com/react-component/menu/pull/655
- [x] (undefined) Circular dependency detected:
node_modules/rc-menu/es/utils/nodeUtil.js -> node_modules/rc-menu/es/index.js -> node_modules/rc-menu/es/Menu.js -> node_modules/rc-menu/es/SubMenu/index.js -> node_modules/rc-menu/es/utils/nodeUtil.js @kiner-tang https://github.com/react-component/menu/pull/655
- [x] (undefined) Circular dependency detected:
node_modules/rc-overflow/es/Overflow.js -> node_modules/rc-overflow/es/RawItem.js -> node_modules/rc-overflow/es/Overflow.js @MuxinFeng https://github.com/react-component/overflow/pull/36
- [x] (undefined) Circular dependency detected:
node_modules/rc-overflow/es/RawItem.js -> node_modules/rc-overflow/es/Overflow.js -> node_modules/rc-overflow/es/RawItem.js @MuxinFeng https://github.com/react-component/overflow/pull/36
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/DecadePanel/DecadeBody.js -> node_modules/rc-picker/es/panels/DecadePanel/index.js -> node_modules/rc-picker/es/panels/DecadePanel/DecadeBody.js @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/DecadePanel/DecadeHeader.js -> node_modules/rc-picker/es/panels/DecadePanel/index.js -> node_modules/rc-picker/es/panels/DecadePanel/DecadeHeader.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/DecadePanel/index.js -> node_modules/rc-picker/es/panels/DecadePanel/DecadeHeader.js -> node_modules/rc-picker/es/panels/DecadePanel/index.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/PanelBody.js -> node_modules/rc-picker/es/utils/dateUtil.js -> node_modules/rc-picker/es/panels/DecadePanel/index.js -> node_modules/rc-picker/es/panels/DecadePanel/DecadeBody.js -> node_modules/rc-picker/es/panels/PanelBody.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/YearPanel/YearBody.js -> node_modules/rc-picker/es/panels/YearPanel/index.js -> node_modules/rc-picker/es/panels/YearPanel/YearBody.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/YearPanel/YearHeader.js -> node_modules/rc-picker/es/panels/YearPanel/index.js -> node_modules/rc-picker/es/panels/YearPanel/YearHeader.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/panels/YearPanel/index.js -> node_modules/rc-picker/es/panels/YearPanel/YearHeader.js -> node_modules/rc-picker/es/panels/YearPanel/index.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-picker/es/utils/dateUtil.js -> node_modules/rc-picker/es/panels/DecadePanel/index.js -> node_modules/rc-picker/es/panels/DecadePanel/DecadeBody.js -> node_modules/rc-picker/es/panels/PanelBody.js -> node_modules/rc-picker/es/utils/dateUtil.js  @kiner-tang https://github.com/react-component/picker/pull/650
- [x] (undefined) Circular dependency detected:
node_modules/rc-tree/es/TreeNode.js -> node_modules/rc-tree/es/utils/treeUtil.js -> node_modules/rc-tree/es/util.js -> node_modules/rc-tree/es/TreeNode.js @MuxinFeng https://github.com/react-component/tree/pull/749
- [x] (undefined) Circular dependency detected:
node_modules/rc-tree/es/util.js -> node_modules/rc-tree/es/TreeNode.js -> node_modules/rc-tree/es/utils/treeUtil.js -> node_modules/rc-tree/es/util.js @MuxinFeng https://github.com/react-component/tree/pull/749
- [x] (undefined) Circular dependency detected:
node_modules/rc-tree/es/utils/treeUtil.js -> node_modules/rc-tree/es/util.js -> node_modules/rc-tree/es/TreeNode.js -> node_modules/rc-tree/es/utils/treeUtil.js @MuxinFeng https://github.com/react-component/tree/pull/749

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Resolved Circular dependency issue in vite, rollup, meteor and microbundle.         |
| 🇨🇳 Chinese |  解决 vite、rollup、meteor、microbundle 等构建工具中遇到的循环依赖问题，并增加相关的检测。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### CODE REVIEW

**Comment by jrr997** [comment]
认领 Alert 和 Checkbox。

**Comment by jrr997** [comment]
PS：任务列表重复了，复制了两遍。

**Comment by kiner-tang** [comment]
> PS：任务列表重复了，复制了两遍。

感谢提醒，已经删掉了重复项

**Comment by jrr997** [comment]
领取 config-provider date-picker descriptions

**Comment by jrr997** [comment]
领取 table

**Comment by kiner-tang** [comment]
@jrr997 记得已经提交的 PR 合并一下 feature，并将 base 改为 feature 哈

**Comment by kiner-tang** [comment]
> > * dropdown
> 
> dropdown 似乎没有循环依赖了，大佬有空帮我瞅瞅呗 @kiner-tang

我看看

**Comment by kiner-tang** [comment]
> > > * dropdown
> > 
> > 
> > dropdown 似乎没有循环依赖了，大佬有空帮我瞅瞅呗 @kiner-tang
> 
> 我看看

确实是这样，估计是那个 PR 顺带改了，我把 dropdown 任务删掉吧

**Comment by kiner-tang** [comment]
@afc163 重构任务看来都有人认领并有 PR 了，没有合并的 PR 麻烦看看，很多是 rc 组件，我们这边没有权限

**Comment by afc163** [comment]
https://github.com/react-component/color-picker/releases/tag/v1.3.0

File: package.json [hunk 1]
```diff
@@ -128,16 +128,16 @@
     "rc-dialog": "~9.1.0",
     "rc-drawer": "~6.2.0",
     "rc-dropdown": "~4.1.0",
-    "rc-field-form": "~1.32.0",
-    "rc-image": "~6.0.0",
+    "rc-field-form": "~1.33.0",
+    "rc-image": "~6.1.0",
     "rc-input": "~1.1.0",
     "rc-input-number": "~8.0.0",
     "rc-mentions": "~2.4.1",
```
Comment (afc163) [inline package.json:135]:
```suggestion
    "rc-mentions": "~2.5.0",
```

File: package.json [hunk 2]
```diff
@@ -150,7 +150,7 @@
     "rc-tabs": "~12.8.1",
```
Comment (afc163) [inline package.json:150]:
```suggestion
    "rc-tabs": "~12.9.0",
```

Comment (afc163) [inline package.json:150]:
```suggestion
    "rc-tabs": "~12.9.0",
```

Comment (afc163) [inline package.json:150]:
```suggestion
    "rc-tabs": "~12.9.0",
```

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**